dartdoc_test/lib/src/extractor.dart

```diff
@@ -11,71 +11,145 @@ final _md = Document(extensionSet: ExtensionSet.gitHubWeb);
     required this.code,
+    this.span,
+    this.imports = const [],
+    this.hasMain = false,
   });
+
+  String get wrappedCode {
+    final fileName = comment.span.file.url?.path.split('/').last;
+
+    final buf = StringBuffer();
+    buf.writeAll(comment.imports, '\n');
+    buf.writeln();
+    buf.writeln('import "$fileName";');
+    buf.writeln();
+    buf.writeln('void main() {');
+    buf.writeAll(code.split('\n').map((l) => '  $l'), '\n');
+    buf.writeln('}');
+    return buf.toString();
+  }
 }
 
-class Extractor {
-  const Extractor();
-
-  void extractFile() {}
-
-  List<DocumentationComment> extractDocumentationComments(ParsedUnitResult r) {
-    final file = SourceFile.fromString(r.content, url: r.uri);
-    final comments = <DocumentationComment>[];
-    r.unit.accept(_ForEachCommentAstVisitor((comment) {
-      if (comment.isDocumentation) {
-        final span = file.span(comment.offset, comment.end);
-
-        // TODO: remove `///` syntax with a better way..
-        var lines = LineSplitter.split(span.text);
-        lines = lines.map((l) => l.replaceFirst('///', ''));
-        comments.add(DocumentationComment(
-          contents: lines.join('\n'),
-          span: span,
-        ));
-      }
-    }));
-    return comments;
+List<DocumentationCodeSample> extractFile(
+  ParsedUnitResult result,
+) {
+  var samples = <DocumentationCodeSample>[];
+  final comments = extractDocumentationComments(result);
+  for (final c in comments) {
+    final extractedSamples = extractCodeSamples(c);
+    samples.addAll(extractedSamples);
   }
+  return samples;
+}
+
+List<DocumentationComment> extractDocumentationComments(ParsedUnitResult r) {
+  final file = SourceFile.fromString(r.content, url: r.uri);
+
+  final imports = getImports(file, r);
+
+  final comments = <DocumentationComment>[];
+  r.unit.accept(_ForEachCommentAstVisitor((comment) {
+    if (comment.isDocumentation) {
+      final span = file.span(comment.offset, comment.end);
+      final content = stripComments(span.text);
 
-  List<DocumentationCodeSample> extractCodeSamples(
-    DocumentationComment comment,
-  ) {
-    final samples = <DocumentationCodeSample>[];
-    final nodes = _md.parse(comment.contents);
-    nodes.accept(_ForEachElement((element) {
-      if (element.tag == 'code' &&
-          element.attributes['class'] == 'language-dart') {
-        var code = '';
-        element.children?.accept(_ForEachText((text) {
-          code += text.textContent;
-        }));
-        if (code.isNotEmpty) {
-          samples.add(DocumentationCodeSample(comment: comment, code: code));
-        }
+      comments.add(DocumentationComment(
+        contents: content,
+        span: span,
+        imports: imports,
+      ));
+    }
+  }));
+  return comments;
+}
+
+Iterable<String> stripleadingWhiteSpace(String comment) {
+  final lines = LineSplitter.split(comment);
+  return lines.map((l) => l.trimLeft());
+}
+
+String stripComments(String comment) {
```

Comment:
Please add a documentation comment what what this does. Ideally with some examples.